frontend/src/app/shared/components/user-notification/user-notification.component.ts

```diff
@@ -69,13 +69,14 @@ export class UserNotificationComponent extends BaseComponent implements OnInit,
     }
 
     readAllNotifications() {
-        this.notificationService.readAllNotifications()
-            .subscribe(() => {
-                this.notifications.map(notification => ({
-                    ...notification,
-                    isRead: true,
-                }));
-            });
+        this.notifications = this.notifications
+            .map((notification) => ({
+                ...notification,
+                isRead: true,
+            }))
+            .filter((n) => !n.isRead);
```

Comment:
Are you sure? As it won't return anything?

Reply:
In general, I can assign an empty array